### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by runk** [avoidwork/tiny-lru#134]
Hello, I want to draw your attention to a memory leak problem that's reported for [maxmind](https://www.npmjs.com/package/maxmind) I maintain:  https://github.com/runk/node-maxmind/issues/740

The only change between "good" and "bad" version of maxmind is the version upgrade of tiny-lru in September. Releases with notes: https://github.com/runk/node-maxmind/releases

Do you have any clues what this could be?

**Comment by avoidwork** [comment]
Yes, a bug in their code.

**Comment by avoidwork** [comment]
I looked at your code briefly, but it's hard to follow where you're using the cache between the modules.

If you look at https://github.com/avoidwork/tiny-lru/compare/11.1.2...11.2.0 the only difference that could cause your behavioral change is making 'has()' public, which historically is known to be a problem if anything external makes an assumption on the boolean it returns without considering lazy expiration.

11.1.2 to 11.2.0 exposes 'has()' and exports the class.

**Comment by avoidwork** [comment]
in 11.1.2 a statement was deleted which may be the cause of the leak if it leaves orphans in ram. it doesn't have test coverage.

https://github.com/avoidwork/tiny-lru/compare/11.1.1...11.1.2#diff-79389319fa7fe3ff2ff6e0d5c97a626b12d8a2c72323e8ad210f43c8f98b1d60L129

**Comment by avoidwork** [comment]
Fix has shipped in 11.2.1.

**Comment by avoidwork** [comment]
Test added at https://github.com/avoidwork/tiny-lru/blob/master/test/lru.js#L124-L130

**Comment by runk** [comment]
Thanks for looking into!
